### PR TITLE
Implementation - Settlement API, issue, cancel, send/ resend notification, delete, edit methods

### PR DESCRIPTION
**Pull request by varunajmera0**
Here is my implementation:-
I updated the invoice file methods like -
  1) issue
  2) delete
  3) send/ resend notification ( - payment link)
  4) edit
  5) cancel ( - payment link)
You can use these methods on payment link.

A new **Settlement API** is done.

### CODE REVIEW

**Comment by captn3m0** [comment]
@varunajmera0 Thanks for the Pull Request! We're getting this reviewed internally. :soon:

File: razorpay/resources/invoice.py [hunk 1]
```diff
@@ -45,3 +45,66 @@ def create(self, data={}, **kwargs):
...
+    def notify_by(self, invoice_id, medium, data={}, **kwargs):
+        """"
+        Send/Resend notifications to customer via email/sms
+
+        Args:
+            invoice_id : Id for trigger notify
+            medium : Medium for triggering notification via email or sms
+
+        Returns:
+            True or False
```
[the remark this answers is not in the document]
Comment (varunajmera0) [reply razorpay/resources/invoice.py]: Okay, I will check and get back on this asap.
Thanks

File: razorpay/resources/invoice.py [hunk 2]
```diff
@@ -45,3 +45,66 @@ def create(self, data={}, **kwargs):
         """
         url = self.base_url
         return self.post_url(url, data, **kwargs)
+
+    def notify_by(self, invoice_id, medium, data={}, **kwargs):
```
[the remark this answers is not in the document]
Comment (varunajmera0) [reply razorpay/resources/invoice.py]: Okay, I will check and get back on this asap.
Thanks

File: razorpay/resources/invoice.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (varunajmera0) [reply razorpay/resources/invoice.py]: @jitendra-1217 done. Please check and let me know if any changes you need.

File: razorpay/resources/invoice.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (varunajmera0) [reply razorpay/resources/invoice.py]: @jitendra-1217 done. Please check and let me know if any changes you need.

File: README.md [hunk 3]
```diff
@@ -87,6 +87,49 @@ that both app title and version are strings.
     client.payment.transfers("<PAYMENT_ID>")
     ```
 
+### Payment Link
+
+- Create payment link
+
+    ```py
+    DATA = {"customer": { "name": "Varun Test",
```
Comment (jitendra-1217) [inline README.md]: Need to fix formatting and sample data in all these snippets.

File: razorpay/resources/invoice.py [hunk 4]
```diff
@@ -45,3 +45,131 @@ def create(self, data={}, **kwargs):
...
+    def edit(self, invoice_id, data={}, **kwargs):
+        """"
+        Update an invoice
+        In draft state all the attributes are allowed.
+
+        Args:
+            invoice_id : Id for delete the invoice
+            data : Dictionary having keys using which invoice have to be updated
+        Returns:
+            {
```
Comment (jitendra-1217) [inline razorpay/resources/invoice.py]: Too verbose comment in code, can just say returns json for invoice entity and link to existing docs containing this sample.

**Comment by jitendra-1217** [comment]
Merging this now.
Will fix formatting and other things in readme in a separate branch myself.


Thanks a lot @varunajmera0  👍

**Comment by amitmahbubani** [comment]
Thanks @varunajmera0 💯

**Comment by varunajmera0** [comment]
thanks, @jitendra-1217 @amitmahbubani @captn3m0

**Pull request merged**